py/htm/encoders/date.py

```diff
@@ -230,7 +230,7 @@ def encode(self, inp, output=None):
       output = SDR(self.dimensions)
     else:
       assert( isinstance(output, SDR) )
-      assert( output.dimensions == self.dimensions )
+      assert( all(x == y for x, y in zip( output.dimensions, self.dimensions )))
```

Comment:
that one was easir to read, why this?

Reply:
This way works even if the dimensions are different types, such as tuple vs list.

This class keeps its dimensions as a tuple, the SDR class keeps its dimensions as a list (b/c of pybind11).